src/workerd/jsg/modules.h

```diff
@@ -267,15 +354,40 @@ class ModuleRegistryImpl final: public ModuleRegistry {
   // we need to be able to search it by path (filename) as well as search for a specific module
   // object by identity. We use a kj::Table!
   struct Entry {
+    using Info = kj::OneOf<ModuleInfo,
+                           kj::ArrayPtr<const char>,
+                           kj::Function<ModuleInfo(Lock&)>>;
+
+    struct Key {
+      const kj::Path& specifier;
+      Type type = Type::BUNDLE;
+
+      Key(const kj::Path& specifier, Type type) : specifier(specifier), type(type) {}
+
+      uint hashCode() const {
+        return kj::hashCode(specifier, type);
```

Comment:
If the above comment is true then both `specifier` and `type` are const and we wouldn't need to recalculate the hash and it could just be part of the struct.